packaging/README.md

```diff
@@ -0,0 +1,145 @@
+  variable `CMAKE_DISTRO_RELEASE`.
+
+`pgmajor`, `pgminor` and `pgpatch`
+: is the major, minor, and patch
+  version of the PostgreSQL the package is built for. These are stored
+  in the CMake variables `PG_VERSION_MAJOR`, `PG_VERSION_MINOR`, and
+  `PG_VERSION_PATCH` respectively.
+
+`relno`
+: is the release number. Typically 1, but if you have to make
+  several releases of the same package, you can change this using the
+  CMake variable `TSDB_RELNO`.
+
+`arch`
+: is the architecture as given by `uname -m`.
+
+The file name for an RPM package is typically
+`<packname>-<version>.<arch>.rpm` and for a Debian package it is
+`<packname>_<version>_<arch>.deb`.
+
+### Package name
+
+The package name is used as prefix for the package file name, but also
+use for some directories in the package. If the major PG version is
+12, the three names under consideration are:
+
+- The loader package name is `timescaledb-loader-postgresql-12`.
+- The OSS package name is `timescaledb-oss-postgresql-12`
+- The TSL package name is `timescaledb-postgresql-12`.
+
+### Version
+
+The version consists of the version number followed by release
+information.
+
+For the Debian package, the format is typically
+`<version>-<relno><distid>~<distrel>` and the `distid` and `distrel`
+is retrieved from `lsb_release`. For Debian it is also possible to add
+an epoch, but we do not use that.
+
+For the RPM package, the name is built as
+`<version>-<relno>%{?dist}`. That is, the traditional `%{?dist}` is
+added to the release number and used as release information. This will
+then be filled in by `rpmbuild`.
+
+### Examples
+
+`timescaledb-postgresql-12-1.7.2-1ubuntu~19.10.amd64.deb`
+: is the package name for an Debian package containing Timescale 1.7.2
+  build for PostgreSQL 12 on Ubuntu 19.10 with a 64-bit architecture.
+
+`timescaledb-postgres-12_1.7.2-dev_1centos~8.2.2004_amd64.rpm`
+: is an RPM package for TimescaleDB 1.7.2 built for PostgreSQL 12 on
+  CentOS 8.2.2004 with a 64-bit architecture. [Example not correct,
+  need to be fixed.]
+
+# Build Dependencies
+
+Depending on the OS, you need different things installed.
+
+# Linux systems
+
+TO build packages on Linux, you need to have the following installed:
+
+- `lsb_release` is used to figure out the distribution id and release.
+- `rpmbuild` to build RPM packages.
+- `dpkg-dev` to build Debian packages.
+
+# Components
+
+The files of the repository is partitioned into four components that
+are used to build the packages.
+
+- `loader` are the main files of the extension, meaning the control
+  file and the loader shared library and also copyright for the loader
+  package.
+- The `basic` component contains all the files related to the open-source
+  package under Apache license.
+- The `full` component contains all the files related to the Timescale
+  package, except the files in the `basic` package.
+- The `testing` component contains all the testing libraries and
+  files.
```

Comment:
Not sure if we need this. Files that do not install with a particular component are put in the "Unspecified" component and by default installed (but not put in a package).